packages/node/src/indexer/fetch.module.ts

```diff
@@ -102,13 +104,24 @@ import { UnfinalizedBlocksService } from './unfinalizedBlocks.service';
       ],
     },
     FetchService,
-    BenchmarkService,
+    IndexingBenchmarkService,
+    {
+      provide: PoiBenchmarkService,
+      useFactory: (nodeConfig: NodeConfig) => {
+        if (nodeConfig.proofOfIndex) {
+          return new PoiBenchmarkService(nodeConfig);
+        }
+        return;
```

Comment:
Doing this means we cant rely on typescript. Maybe we would be better off making the constructor of PoiBenchmarkService a no-op if proof of index is disabled